tests/src/unit_v2_tests.cpp

```diff
@@ -267,4 +274,136 @@ TEST(stronk_units_v2, can_override_units)
+    auto hr = make<std::ratio<60UL * 60, 1>, seconds, double>(1.0);
+    auto ms = make<std::milli, seconds, double>(1.0);
+    auto d = make<std::ratio<60UL * 60 * 24, 1>, seconds, double>(1.0);
+
+    static_assert(std::same_as<decltype(m / s), meters_per_second::value<double>>);
+    static_assert(std::same_as<decltype(m / hr), meters_per_second::scaled_t<std::ratio<1, 3600>>::value<double>>);
+    static_assert(std::same_as<decltype(km / hr), meters_per_second::scaled_t<std::ratio<5, 18>>::value<double>>);
+    static_assert(std::same_as<decltype(km / d), meters_per_second::scaled_t<std::ratio<5, 432>>::value<double>>);
+    static_assert(std::same_as<decltype(um / ms), meters_per_second::scaled_t<std::ratio<1, 1000>>::value<double>>);
+    static_assert(std::same_as<decltype(ms / um), seconds_per_meter::scaled_t<std::ratio<1000, 1>>::value<double>>);
+
+    [[maybe_unused]]
+    auto per_m = 1 / m;
+    [[maybe_unused]]
+    auto per_km = 1 / km;
+    [[maybe_unused]]
+    auto per_um = 1 / um;
+
+    static_assert(std::same_as<decltype(per_m), per_meter::value<double>>);
+    static_assert(std::same_as<decltype(per_km), per_meter::scaled_t<std::ratio<1, 1'000>>::value<double>>);
+    static_assert(std::same_as<decltype(per_um), per_meter::scaled_t<std::ratio<1'000'000, 1>>::value<double>>);
+
+    static_assert(std::same_as<decltype(m * per_m), double>);
+    static_assert(std::same_as<decltype(km * per_km), double>);
+    static_assert(std::same_as<decltype(um * per_um), double>);
+}
+
+TEST(stronk_units_v2, scales_are_applied_correctly_when_converted)
+{
+    auto m = make<meters, double>(2.0);
+    using km_t = meters::scaled_t<std::kilo>;
+    using um_t = meters::scaled_t<std::micro>;
+
+    auto km = m.to<std::kilo>();
+    auto expected_km = make<km_t>(2.0 / 1'000.0);
+    EXPECT_EQ(km, expected_km);
+
+    auto um = m.to<std::micro>();
+    auto expected_um = make<um_t>(2'000'000.0);
+    EXPECT_EQ(um, expected_um);
+
+    um = make<std::kilo, meters>(3.0).to<std::micro>();
+    expected_um = make<um_t>(3'000'000'000.0);
+    EXPECT_EQ(um, expected_um);
+
+    km = make<um_t>(4.0).to<std::kilo>();
+    expected_km = make<km_t>(4.0 / 1'000'000'000.0);
+    EXPECT_EQ(km, expected_km);
+
+    km = make<km_t>(8.0);
+    auto s = make<seconds>(7200.0);
+    auto km_per_s = km / s;
+    auto expected_speed = unit_scaled_value_t<std::kilo, meters_per_second, double>(0.00111111111111);
+
+    EXPECT_EQ(km_per_s, expected_speed);
+
+    using hr_t = seconds::scaled_t<std::ratio<60UL * 60>>;
+    using km_per_hr_t = meters_per_second::scaled_t<std::ratio<1000, 60UL * 60>>;
+
+    auto hr = s.to<std::ratio<60UL * 60>>();
+    auto expected_hr = make<hr_t>(2.0);
+    EXPECT_EQ(hr, expected_hr);
+
+    auto km_per_hr = km / hr;
+    auto expected_km_per_hr = unit_value_t<km_per_hr_t, double>(4.0);
+
+    EXPECT_EQ(km_per_hr, expected_km_per_hr);
+    auto converted_to_km_per_hr = km_per_s.to<std::ratio<5, 18>>();
```

Comment:
I'm not sure I follow this ratio..

Reply:
Haha I'll change it to the normal 1000/3600